www-apps/radicale/radicale-2.1.8.ebuild

```diff
@@ -0,0 +1,68 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="6"
+
+PYTHON_COMPAT=( python{3_4,3_5,3_6} )
+
+inherit eutils distutils-r1 user
+
+MY_PN="Radicale"
+MY_P="${MY_PN}-${PV}"
+
+DESCRIPTION="A simple CalDAV calendar server"
+HOMEPAGE="http://radicale.org/"
+SRC_URI="mirror://pypi/${MY_PN:0:1}/${MY_PN}/${MY_P}.tar.gz"
+
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="+bcrypt"
+
+RDEPEND=">=dev-python/vobject-0.9.5[${PYTHON_USEDEP}]
+	bcrypt? ( dev-python/passlib[bcrypt,${PYTHON_USEDEP}] )"
+
+S=${WORKDIR}/${MY_P}
+
+RDIR=/var/lib/radicale
+LDIR=/var/log/radicale
+
+pkg_setup() {
+	enewgroup radicale
+	enewuser radicale -1 -1 ${RDIR} radicale
+}
+
+python_install_all() {
+	rm README* || die
+
+	# init file
+	newinitd "${FILESDIR}"/radicale.init.d radicale
+
+	# directories
+	diropts -m0750
+	dodir ${RDIR}
+	fowners radicale:radicale ${RDIR}
+	keepdir ${RDIR}
+	diropts -m0755
+	dodir ${LDIR}
+	fowners radicale:radicale ${LDIR}
+	keepdir ${LDIR}
+
+	# config file
+	insinto /etc/${PN}
+	doins config logging
+
+	# fcgi and wsgi files
+	exeinto /usr/share/${PN}
```

Comment:
Executable files don't belong in `/usr/share`.

Reply:
Here I’m not sure what the right answer is (and again the existing ebuild put them there which I just copied). I’m not sure if `radicale.fcgi` is useful to invoke on its own; I vaguely remember back when I used it I think I ran it through `spawn-fcgi` instead, not invoking it on its own, though I’m not too familiar with the business of how Python does FastCGI. Similarly executing `radicale.wsgi` on its own accomplishes absolutely nothing; it’s meant to be run through a WSGI container like uWSGI. So I don’t think they should be in `$PATH`. Maybe `/usr/libexec`? What do you think? I also discovered that, with uWSGI, `radicale.wsgi` doesn’t actually need execute permission, but I have no idea if that’s true with other containers. Thoughts? Should I leave it executable or make it not?

Reply:
Ok, is this something users put in their scripts via full path? If they do, then it's probably not worth the effort to move it right now.

As for executable bits, I think usually they aren't necessary but again, it depends on how it's used.

Reply:
> Ok, is this something users put in their scripts via full path? If they do, then it's probably not worth the effort to move it right now.

I have Radicale deployed via uWSGI, and my config file contains `--wsgi-file /usr/share/radicale/radicale.wsgi`. So yes.

> As for executable bits, I think usually they aren't necessary but again, it depends on how it's used.

If there’s any possibility of them needed, I would prefer to keep them set, to avoid running into issues like <http://redmine.lighttpd.net/issues/2374>—that was the opposite problem, but conceptually similar in that it was a file permission issue and required fixing file permissions by hand after every upgrade, which was annoying. Are you OK with leaving them executable? If any changes were to be made at all, I *think* it would be that `radicale.wsgi` could be non-executable, but `radicale.fcgi` would still need to be executable.